src/vs/editor/contrib/rtv/RTVSynthDisplay.ts

```diff
@@ -303,48 +397,389 @@ export class RTVSynthDisplayBox implements IRTVDisplayBox{
 			cellContent = renderedText.element;
 		}
 
-		// `this._cellDictionary` is empty here
-		if (this.lineNumber === elmt.controllingLineNumber) {
-			const name = elmt.vname!;
-			if (name in this._cellDictionary) {
-				this._cellDictionary[name].push(cell);
+
+		// Remove any existing content
+		cell.childNodes?.forEach((child) => cell.removeChild(child));
+
+		if (elmt.editable) {
+			// make the TableCellElement `td` editable if applicable
+			cell.contentEditable = 'true';
+		}
+
+		const outputVars = new Set(this.outputVars);
+
+		// Add the new content
+		cell.appendChild(cellContent);
+
+		if (!header && outputVars.has(elmt.vname!)) {
+			this.addListeners(cell);
+		}
+
+		return cell;
+	}
+
+	// ------------
+	// utility functions
+	// ------------
+
+	private select(node: Node) {
+		let selection = window.getSelection()!;
+		let range = selection.getRangeAt(0);
+		range.selectNodeContents(node);
+		selection.removeAllRanges();
+		selection.addRange(range);
+		this.updateCursorPos(range, node as HTMLElement);
+	}
+
+	private findParentRow(cell: HTMLElement): HTMLTableRowElement {
```

Comment:
Can we set `id`s for the table/rows/cells so that we can find them without having to navigate the DOM?